DataDog/datadog-lambda-js
# Initialise customer's copy of dd-trace

### What does this PR do?
Overrides the require of dd-trace, to prefer using the version in the handler's node_modules file.
It also bumps the dd-trace version, the latest version includes a feature which

### Motivation

Makes it possible for a customer to use their own version of dd-trace with the layer. Currently, if a customer installs their own copy of dd-trace via node_modules, but uses the redirected handler, the layer will init it's own version of dd-trace, not the one the customer's code is using. This results in a missing or incomplete trace.

### Testing Guidelines

I manually tested this, it's difficult to verify this behavior in an integration test because it involves a dual dd-trace setup.

### Additional Notes



### Types of Changes


### Check all that apply
